x-pack/plugins/enterprise_search/public/applications/app_search/app_logic.ts

```diff
@@ -7,17 +7,24 @@
 import { kea, MakeLogicType } from 'kea';
 
 import { IInitialAppData } from '../../../common/types';
+import { IConfiguredLimits, IAccount, IRole } from './types';
 
 export interface IAppValues {
   hasInitialized: boolean;
+  ilmEnabled: boolean;
+  configuredLimits: IConfiguredLimits;
+  account: IAccount;
+  myRole: IRole;
 }
 export interface IAppActions {
-  initializeAppData(props: IInitialAppData): void;
+  initializeAppData(props: IInitialAppData): Required<IInitialAppData>;
```

Comment:
Another comment I left but got lost in the ether. Why do you need the Required utility type here?

Reply:
[IInitialAppData has `?:` optional markers on everything](https://github.com/elastic/kibana/blob/master/x-pack/plugins/enterprise_search/common/types/index.ts#L16-L27) because that's one of the possible realities of our API endpoint - it can return an empty obj or missing fields under some conditions (error connecting, or no access, etc.).

Part of that is an artifact of this being a common type between our server files and our public files - our server logic handles the empty obj scenario for us by disallowing access to the plugin or throwing an Error Connecting status. So by the time `IInitialAppData` reaches the public/client-side files, we can be confident that all the obj keys exists - so we use `Required<>` here.

Hopefully that makes sense!

Reply:
Perfect sense, thank you for explaining.